code/modules/lighting/lighting_turf.dm

```diff
@@ -88,6 +88,11 @@
 			else //If fulltile and opaque, then the whole tile blocks view, no need to continue checking.
 				directional_opacity = ALL_CARDINALS
 				break
+	else
+		for(var/atom/elem in src.contents)
+			for(var/datum/light_source/source in elem.light_sources)
+				if(!source?.effect_str)
+					source.force_update()
```

Comment:
My gut is a signal is better here (cleaner and more performant, less things to filter)
Does this cover all cases? IE if a source passes through this space and is cut off, would it remain cutoff after  this proc is called?